include/utils/security_utils.php

```diff
@@ -105,36 +105,21 @@ function get_user_disallowed_modules($user_id, &$allowed)
 // grabs client ip address and returns its value
 function query_client_ip()
 {
-	global $_SERVER;
-	$clientIP = false;
-	if(!empty($GLOBALS['sugar_config']['ip_variable']) && !empty($_SERVER[$GLOBALS['sugar_config']['ip_variable']])){
-		$clientIP = $_SERVER[$GLOBALS['sugar_config']['ip_variable']];
-	}else if(isset($_SERVER['HTTP_CLIENT_IP']))
-	{
-		$clientIP = $_SERVER['HTTP_CLIENT_IP'];
-	}
-	elseif(isset($_SERVER['HTTP_X_FORWARDED_FOR']) AND preg_match_all('#\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}#s', $_SERVER['HTTP_X_FORWARDED_FOR'], $matches))
-	{
-		// check for internal ips by looking at the first octet
-		foreach($matches[0] AS $ip)
-		{
-			if(!preg_match("#^(10|172\.16|192\.168)\.#", $ip))
-			{
-				$clientIP = $ip;
-				break;
-			}
-		}
-
-	}
-	elseif(isset($_SERVER['HTTP_FROM']))
-	{
-		$clientIP = $_SERVER['HTTP_FROM'];
-	}
-	else
-	{
-		$clientIP = $_SERVER['REMOTE_ADDR'];
-	}
-	return $clientIP;
+    if (isset($_SERVER['HTTP_CLIENT_IP'])) {
+        return $_SERVER['HTTP_CLIENT_IP'];
+    } elseif (isset($_SERVER['HTTP_X_FORWARDED_FOR'])) {
+        return $_SERVER['HTTP_X_FORWARDED_FOR'];
+    } elseif (isset($_SERVER['HTTP_X_FORWARDED'])) {
+        return $_SERVER['HTTP_X_FORWARDED'];
+    } elseif (isset($_SERVER['HTTP_FORWARDED_FOR'])) {
+        return $_SERVER['HTTP_FORWARDED_FOR'];
+    } elseif (isset($_SERVER['HTTP_FORWARDED'])) {
+        return $_SERVER['HTTP_FORWARDED'];
+    } elseif (isset($_SERVER['REMOTE_ADDR'])) {
+        return $_SERVER['REMOTE_ADDR'];
+    } else {
+        return false;
```

Comment:
Can we throw an error and/or log a warning if and when there is no any chance to get the client IP?